hal/mps2/cmsis_nvic.h

```diff
@@ -0,0 +1,47 @@
+/* MPS2 CMSIS Library
+*
+* Copyright (c) 2006-2018 ARM Limited
+* SPDX-License-Identifier: BSD-3-Clause
```

Comment:
The BSD-3 license is problematic. Need to check if there is another way.

Reply:
I found this [cmsis_nvic.h](https://github.com/ARMmbed/mbed-os/blob/master/targets/TARGET_ARM_SSG/TARGET_CM3DS_MPS2/device/cmsis_nvic.h) for arm cortex M3 which is license under apache 2.0, and it work without needing any modification.
But I'm not sure am I able to just simply copied it into our repo, and do some slight modification ?

Reply:
And for those files that already included spdx license identifier with apache 2.0 in a block of comment at the top of the file, I still have to add a spdx license identifier at the topmost of the file ?

Reply:
Yes, you can simply copy that in and modify it. Thank you!

Reply:
> And for those files that already included spdx license identifier with apache 2.0 in a block of comment at the top of the file, do I still need to add a spdx license identifier at the topmost of the file ?

No it's okay. I just missed that it's already there.